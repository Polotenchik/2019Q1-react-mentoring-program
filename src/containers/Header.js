import React from 'react';

export const Header = () => {
    return (
        <header className={'header'} style={{ "background": "#000 url('img/netflix-bg.jpg')"}}>
            <div className={'header-container'}>
                Search
            </div>
        </header>
    );
};