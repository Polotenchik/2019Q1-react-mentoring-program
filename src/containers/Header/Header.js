import React from 'react';
import { TopBarSection } from '../../components';
import { Search, MovieDescription } from '../index';

export const Header = (props) => {

    return (
        <header className='header' style={{ "background": "#000 url('img/netflix-bg.jpg')" }}>
            <div className='header-container'>
                {   (props.headerType === 'info') 
                    ? (<>
                        <TopBarSection headerType={ props.headerType } />
                        <MovieDescription movie={ props.id } />
                    </>)
                    : (<>
                        <TopBarSection />
                        <Search />
                    </>)}
            </div>
        </header>
    );
};