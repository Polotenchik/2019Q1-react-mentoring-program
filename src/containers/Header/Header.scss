.header {
    min-height: 350px;
    width: inherit;
}
  
  .header-container {
    width: $headerContainerWidth;
    margin: 0 auto;
    padding: 20px 0;
  }