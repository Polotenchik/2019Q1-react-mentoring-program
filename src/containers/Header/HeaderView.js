import React from 'react';
import { TopBarSection } from '../../components';
import Search from '../Search/Search';
import { MovieDescription } from '../index';

const HeaderView = ({ movieMode, currentMovie, onOpenSearch }) => (
    <header className='header' style={{ background: "#000 url('img/netflix-bg.jpg')" }}>
        <div className='header-container'>
            <TopBarSection searchButtonClick={ onOpenSearch } movieMode={ movieMode } />
            <Search />
            <MovieDescription info={ currentMovie } active={ movieMode } />
        </div>
    </header>
);

HeaderView.defaultProps = {
  filmMode: false,
  currentFilm: {},
};

export default HeaderView;