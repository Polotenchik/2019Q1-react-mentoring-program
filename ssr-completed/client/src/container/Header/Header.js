import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { Route, Switch, withRouter } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Cockpit from '../../components/Cockpit/Cockpit';
import Search from '../../components/Search/Search';
import {
  toggleSearchBy, searchButtonHandler, toggleSortBy, fetchMovies, saveTerm, toggleSortOrder,
} from '../../store';
import {getUrl} from '../../constants';

class Header extends Component {
  state={
    term: '',
  };

  inputChangeHandler = ({ target: { value } }) => {
    this.setState({
      term: value,
    });
  }

  toggleSortDirectionHandler = () => {
    const {
      sortOrder, toggleSortOrder, fetchMovies, term, searchBy, sortBy, offset, moviesPerPage,
    } = this.props;
    const sorted = sortOrder === 'asc' ? 'desc' : 'asc';
    toggleSortOrder(sorted);
    const url =  getUrl(searchBy, sortBy, sorted, term, offset, moviesPerPage);
    fetchMovies(url);
  }

  toggleSortByHandler = (sortBy) => {
    const {
      toggleSortBy, fetchMovies, searchBy, sortOrder, term, offset, moviesPerPage,
    } = this.props;
    toggleSortBy(sortBy);
    const url =  getUrl(searchBy, sortBy, sortOrder, term, offset, moviesPerPage);
    fetchMovies(url);
  }

  formSubmitHandler = (e) => {
    e.preventDefault();
    const { term } = this.state;
    const {
      saveTerm, fetchMovies, searchBy, sortBy, sortOrder, offset, moviesPerPage, history,
    } = this.props;
    saveTerm(term);

   // location.pathname === '/search' 
  
  //  : history.push(`/search?searchBy=${searchBy}&sortBy=${sortBy}&sortOrder=${sortOrder}&search=${term}&offset=${offset}&limit=${moviesPerPage}`);
  
    history.push(`/s/${term.replace(/\s+/,'-')}`);
   // history.push(`/search?searchBy=${searchBy}&sortBy=${sortBy}&sortOrder=${sortOrder}&search=${term}&offset=${offset}&limit=${moviesPerPage}`);
    const url =  getUrl(searchBy, sortBy, sortOrder, term, offset, moviesPerPage);
    fetchMovies(url);
  }

  render() {
    const {
      total,
      searchBy,
      sortBy,
      sortOrder,
      toggleSearchBy,
    } = this.props;

    return (
      <Fragment>
        <HeaderWrapper>
          <Search searchBy={searchBy} inputChangeHandler={this.inputChangeHandler} toggleSearchBy={toggleSearchBy} formSubmitHandler={this.formSubmitHandler} />
          <Cockpit sortBy={sortBy} sortOrder={sortOrder} toggleSortBy={this.toggleSortByHandler} toggleSortDirection={this.toggleSortDirectionHandler} filmsCount={total} />
        </HeaderWrapper>
      </Fragment>
    );
  }
}

export default withRouter(connect(({ movies, movie, search }) => ({
  total: movies.total,
  activeFilm: movie.activeFilm,
  offset: search.offset,
  moviesPerPage: search.moviesPerPage,
  searchBy: search.searchBy,
  sortBy: search.sortBy,
  sortOrder: search.sortOrder,
  term: search.term,
}),
{
  toggleSearchBy, searchButtonHandler, toggleSortBy, fetchMovies, saveTerm, toggleSortOrder,
})(Header));

Header.propTypes = {
  sortOrder: PropTypes.string,
  total: PropTypes.number,
  sortBy: PropTypes.string,
  searchBy: PropTypes.string,
  toggleSearchBy: PropTypes.func,
  toggleSortBy: PropTypes.func,
  toggleSortOrder: PropTypes.func,
  fetchMovies: PropTypes.func,
  saveTerm: PropTypes.func,
  activeFilm: PropTypes.shape(),
  term: PropTypes.string,
  offset: PropTypes.number,
  moviesPerPage: PropTypes.number,
  history: PropTypes.shape(),
};

const HeaderWrapper = styled.header`
  flex: 0 0 auto; 
`;
