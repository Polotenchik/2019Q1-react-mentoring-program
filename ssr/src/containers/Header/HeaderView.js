import React from 'react';
import PropTypes from 'prop-types';
import { TopBarSection } from '../../components';
import Search from '../Search/Search';
import MovieDescription from '../MovieDescription/MovieDescription';

export class HeaderView extends React.Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    if (this.props.match.params.id) {
      this.props.onFetchMovieById(this.props.match.params.id);
    }
  }

  render() {
    return (
            <header className='header' style={{ background: "#000 url('img/netflix-bg.jpg')" }}>
                <div className='header-container'>
                    <TopBarSection searchButtonClick={ this.props.onOpenSearch } movieMode={ this.props.movieMode }/>
                    <Search />
                    <MovieDescription />
                </div>
            </header>
    );
  }
}

HeaderView.defaultProps = {
  filmMode: false,
  currentFilm: {},
};
