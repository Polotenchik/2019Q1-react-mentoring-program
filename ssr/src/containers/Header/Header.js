import { connect } from 'react-redux';
import { HeaderView } from './HeaderView';
import { openSearch } from '../../redux/mode/mode.actions';
import { fetchMovieById } from '../../redux/movies/movies.actions';

const mapStateToProps = store => ({
  movieMode: store.mode.movie,
  currentMovie: store.movies.currentMovie,
});

const mapDispatchToProps = dispatch => ({
  onOpenSearch() {
    dispatch(openSearch());
  },
  onFetchMovieById(id) {
    dispatch(fetchMovieById(id));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(HeaderView);
