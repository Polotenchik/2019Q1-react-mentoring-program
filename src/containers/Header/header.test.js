import React from 'react';
import { Header } from './Header';
import { shallow } from 'enzyme';

describe('<Header />', () => {
  describe('Rendering', () => {
    it('should render default header', () => {
      const wrapper = shallow(<Header />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});