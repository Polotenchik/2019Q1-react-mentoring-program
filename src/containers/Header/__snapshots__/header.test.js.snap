// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Rendering should render default header 1`] = `
<header
  className="header"
  style={
    Object {
      "background": "#000 url('img/netflix-bg.jpg')",
    }
  }
>
  <div
    className="header-container"
  >
    <TopBarSection />
    <Search />
  </div>
</header>
`;
